### PR TITLE
Cherry-picked SoundRemapper

### PR DESCRIPTION
**Pull request by 7kasper**
Removes the legacy Soundremapper for a new system more like the idRemappers but with Strings.

### CODE REVIEW

**Comment by 7kasper** [comment]
Will fix enum madness and legacysoundstring tomorrow. Sleep tight! :P

File: src/protocolsupport/zplatform/PlatformUtils.java [hunk 1]
```diff
@@ -61,5 +61,5 @@
 	public String getPrependerHandlerName();
 
 	public void setFraming(ChannelPipeline pipeline, IPacketSplitter splitter, IPacketPrepender prepender);
-
-}
+
```
Comment (Shevchik) [inline src/protocolsupport/zplatform/PlatformUtils.java]: whitespace

File: src/protocolsupport/utils/Utils.java [hunk 2]
```diff
@@ -127,6 +130,20 @@ public static int ceilToBase(int number, int base) {
 	public static boolean isTrue(Boolean b) {
 		return (b != null) && b;
 	}
+	
+	public static int TIntObjectHashMapGetKeyFromValue(TIntObjectHashMap<?> map, Object fromValue) {
```
Comment (Shevchik) [inline src/protocolsupport/utils/Utils.java]: not used?

File: src/protocolsupport/protocol/utils/minecraftdata/SoundData.java [hunk 3]
```diff
@@ -20,5 +21,9 @@
 	public static String getNameById(int id) {
 		return idToName.get(id);
 	}
+	
+	public static int getIdByName(String name) {
```
Comment (Shevchik) [inline src/protocolsupport/protocol/utils/minecraftdata/SoundData.java]: not used?

Comment (7kasper) [inline src/protocolsupport/protocol/utils/minecraftdata/SoundData.java]: Meh

File: src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_10_11_12/WorldSound.java [hunk 4]
```diff
@@ -6,16 +6,21 @@
...
 import protocolsupport.utils.recyclable.RecyclableCollection;
+import protocolsupport.utils.recyclable.RecyclableEmptyList;
 import protocolsupport.utils.recyclable.RecyclableSingletonList;
 
 public class WorldSound extends MiddleWorldSound {
 
 	@Override
 	public RecyclableCollection<ClientBoundPacketData> toData(ProtocolVersion version) {
+		String soundname = SoundRemapper.getSoundName(version, id);
+		if (soundname == null || soundname == "") {
```
Comment (Shevchik) [inline src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_10_11_12/WorldSound.java]: apache commons lang3 StringUtils::isEmpty

**Comment by 7kasper** [comment]
WAIt

**Pull request merged**